platform/wayland/cog-platform-wl.c

```diff
@@ -980,11 +1029,21 @@ touch_on_down(void              *data,
               wl_fixed_t         x,
               wl_fixed_t         y)
 {
-    CogWlPlatform *platform = data;
-    CogWlDisplay  *display = platform->display;
+    CogWlSeat    *seat = data;
+    CogWlDisplay *display = seat->display;
 
-    display->seat.touch.surface = surface;
-    display->seat.keyboard.serial = serial;
+    if (touch != seat->touch_obj) {
+        g_critical("%s: Got touch %p, expected %p.", G_STRFUNC, touch, seat->touch_obj);
+        return;
+    }
+
+    seat->display->seat_default = seat;
+
+    CogWlPlatform *platform =
+        wl_surface_get_user_data(surface); // Platform is attached the context of the wl_surface. Check create_window();
+    seat->touch_target = platform;
```

Comment:
I find this counterintuitive; the target of events should be a CogView, or at least a `wl_surface`. Maybe the `*_target` variables need a better name, or the `CogWlPlatform` has no business being the “target”?

I think for now we can use `cog_platform_get_default()` as suggested above, and introduce the `*_target` variables in the future when/if needed. How does that sound to you?

Reply:
DITTO. This is justified in https://github.com/Igalia/cog/pull/623#discussion_r1377704180 but again it is wrong to add changes for problems that are still not in the code yet. I will update the PR according your suggestions.

Reply:
PR updated.